### DISCUSSION TITLE
🗳 User poll: What browsers/environments you need to support?

### DISCUSSION THREAD

**Discussion by hmdne** [opal/opal#2314] [General]
Well, the title says all. We'd like to know how much compatibility we can possibly break in the future releases. The Web moves fast and our codebase is insanely compatible (most of the time only small patches will be needed), but breaking some of it may be worthy so we can implement some of the more recent ideas (see: #2231).

This isn't limited to web browsers, but also other environments like maybe GJS, JScript, ECMAScript versions, anything.

As of now, we say that we support X and X-1 of major web browsers and IE11. This is a close estimation only. IE11 support broke in the Opal 1.2 release, was accidentally fixed for 1.1 and didn't work before.

For me, it's currently ES5 syntax-wise, because opal-optimizer depends on RKelly, which doesn't support anything higher. This is of course fixable in various ways.

Some mobile phones still in use have a very old version of WebView, we may consider this important, as some phones can't even upgrade it, and in the future we may want to use it (here though, Babel can come into rescue).

**Comment by janbiedermann** [comment]
For my purposes:
node LTS and browsers LTS
for mobiles: chrome and firefox, supporting older phones/browser doesnt make much sense for my purposes, because of lower performance, maybe 2 or 3 generations looking back

But also please consider, modern bundlers can handle compatibility by parsing and rebuilding js according to selected target:
https://esbuild.github.io/api/#target
https://esbuild.github.io/content-types/#javascript
Having a fairly recent target, older devices/environments can be supported by using such bundler.

**Comment by tobymao** [comment]
18xx.games uses esbuild targeting es2019

https://github.com/tobymao/18xx/blob/master/lib/assets.rb#L241

**Comment by al6x** [comment]
No need to support old browsers. It could be supported as extra build step with babel. 

`opal -> modern js` + optional build step, if you need to target old browsers `modern js -> babel -> old js`
